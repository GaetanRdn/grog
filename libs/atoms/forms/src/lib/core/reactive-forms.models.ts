export type OnChangeFn<ValueType> = (value: ValueType | null) => void;
export type OnTouchedFn = () => void;
