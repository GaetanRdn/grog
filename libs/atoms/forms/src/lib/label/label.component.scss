@use '../../styles/src/lib/utils';

@mixin adr-label($config) {
  adr-label {
    @include utils.set-typography($config, label);
  }
}
