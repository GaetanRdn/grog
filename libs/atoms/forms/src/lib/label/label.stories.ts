import { LabelComponent } from '@grorg/atoms-forms';
import { Meta, Story } from '@storybook/angular';

export default {
  title: 'atoms/forms/label',
  component: LabelComponent,
  parameters: {
    jest: ['label.component.spec.ts'],
  },
} as Meta<LabelComponent>;

const BasicTemplate: Story<LabelComponent> = (args) => ({
  template: `<adr-label>{{ ngContent }}</adr-label>`,
  props: { ...args },
});

export const Default = BasicTemplate.bind({});
Default.args = {
  ngContent: 'Un label',
};
