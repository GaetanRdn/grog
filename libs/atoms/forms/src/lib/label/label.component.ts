import { CommonModule } from '@angular/common';
import { Component, NgModule } from '@angular/core';

@Component({
  selector: 'adr-label',
  template: `<ng-content></ng-content>`,
})
export class LabelComponent {}

@NgModule({
  declarations: [LabelComponent],
  imports: [CommonModule],
})
export class LabelModule {}
