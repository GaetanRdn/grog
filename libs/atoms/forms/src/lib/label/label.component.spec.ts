import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { LabelComponent } from './label.component';
import { TemplateLookup } from '@grorg/tests';

describe('LabelComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LabelComponent, HostComponent],
    });

    templateLookup = new TemplateLookup<HostComponent>(
      TestBed.createComponent(HostComponent)
    );
  });

  test.each(['toto', 'test'])(
    'should create with content %s',
    (label: string) => {
      // GIVEN
      templateLookup.hostComponent.label = label;

      // WHEN
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <gro-label>{{ label }}</gro-label>`,
})
class HostComponent {
  public label!: string;
}
