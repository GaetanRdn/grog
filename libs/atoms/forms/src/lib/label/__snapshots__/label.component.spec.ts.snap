// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelComponent should create with content test 1`] = `
<adr-label>
  test
</adr-label>
`;

exports[`LabelComponent should create with content toto 1`] = `
<adr-label>
  toto
</adr-label>
`;
